### PR TITLE
step1

### PR DESCRIPTION
**Pull request by ShivanshSinghnnovative**

### CODE REVIEW

File: src/composables/loginsignup.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            await store.dispatch('login', {
+                email: loginUserDetails.email,
+                password: loginUserDetails.password,
+            })
+            if (!loginErr.value) {
+                const userData = {
+                    email: loginUserDetails.email,
+                    password: loginUserDetails.password,
+                };
+                localStorage.setItem('user', JSON.stringify(userData));
```
Comment (PiyushInnotech) [inline src/composables/loginsignup.js]: use userDetails

File: src/composables/loginsignup.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    })
+
+    const loginAccount = async () => {
+        try {
+            await store.dispatch('login', {
+                email: loginUserDetails.email,
+                password: loginUserDetails.password,
+            })
+            if (!loginErr.value) {
+                const userData = {
```
Comment (PiyushInnotech) [inline src/composables/loginsignup.js]: get userDetails from response.

File: src/composables/loginsignup.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    const isLoading = ref(false)
+    const isSubmitted = ref(false)
+    const hidePassword = ref(false)
+    const signUpErr = ref('')
+    const passwordCheck = ref('')
+    const emailCheck = ref('')
+
+    const userExist = ref('')
+
+    store.dispatch('fetchRoles').then(() => {
```
Comment (PiyushInnotech) [inline src/composables/loginsignup.js]: use onMounted and computed property

File: src/composables/loginsignup.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                        })
+                        if (!signErr.value) {
+                            isSubmitted.value = true
+                        }
+                        else {
+                            userExist.value = "Email already exist. Try another email"
+                        }
+                    } catch (error) {
+                        console.log(error)
+                    } finally {
```
Comment (PiyushInnotech) [inline src/composables/loginsignup.js]: remove finally

File: src/composables/loginsignup.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+    const signUser = reactive({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        roleId: null,
+    })
+    const isLoading = ref(false)
+    const isSubmitted = ref(false)
```
Comment (PiyushInnotech) [inline src/composables/loginsignup.js]: change name to something like openSuccesModal

File: src/pages/loginPage.vue [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+const clearLoginError = () => {
+    store.commit('clearLoginError');
+};
+onMounted(() => {
+    clearLoginError();
+})
+
+
+
+
```
Comment (PiyushInnotech) [inline src/pages/loginPage.vue]: too much gap and also check format

File: src/composables/loginsignup.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    onMounted(async () => {
+        try {
+            const rolesData = await store.dispatch('fetchRoles');
+            roles.value = rolesData;
+        } catch (error) {
+            console.error('Error fetching roles:', error);
+        }
+    });
+
+    const rolesData = computed(() => {
```
Comment (yuvrajInno) [inline src/composables/loginsignup.js:68]: there are 2 roles data can't we use 1 @ShivanshSinghnnovative

**Review by yuvrajInno — CHANGES_REQUESTED** [review]

**Pull request merged**